src/containers/main/index.js

```diff
@@ -1,87 +1,57 @@
-import React, { PureComponent } from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import { withRouter } from 'react-router'
-
-import { fetchAllTheToots } from '../../utils/api'
 import { connect } from 'react-redux'
 import { translate } from 'react-i18next'
 
-import { addToots as addTootsAction } from '../../actions/toots'
-
-import Timeline from '../../components/timeline'
-import Styles from './styles'
+import Timeline from '../timeline'
+import { saveLayout as saveLayoutAction } from '../../actions/config'
 
-class Main extends PureComponent {
-	async componentDidMount () {
-		const { user, addToots } = this.props
+import './index.css'
 
-		if (user) {
-			try {
-				const [home, local, federated] = await fetchAllTheToots(user.oauth)
-				addToots({
-					home,
-					local,
-					federated
-				})
-			} catch (e) {
-				console.info("Handle error properly", e)
-			}
-		}
-	}
-
-	render () {
-		const {
-			t,
-			tootsHome,
-			tootsLocal,
-			tootsFederated
-		} = this.props
+const components = {
+  timeline: Timeline
+}
 
-		const isLoading = Boolean(tootsHome.length) === false
+class Main extends Component {
 
-		if (isLoading) {
-			return <div>
-				{t('loading')}
-			</div>
-		}
+  render () {
+    const { layout, currentAccount, match } = this.props
+    let columns = [<Timeline type={match.params.component} accountUrl={currentAccount.user.url} />]
+    layout.map(column => {
```

Comment:
Do not use `map` if you don't return inside :)

Maybe you could take advantage of destructuring here :

```js
const columns = [
    <Timeline type={match.params.component} accountUrl={currentAccount.user.url} />,
    ...layout.map((column) => {
        const Component = components[column.component]
        return <Component {...column} />
    })
]
```

Or even 

```js
const columns = [
    <Timeline type={match.params.component} accountUrl={currentAccount.user.url} />
].concat(layout.map((column) => {
    const Component = components[column.component]
    return <Component {...column} />
})
```

Whatever floats your ⛵